docs/spec_draft.md

```diff
@@ -2963,6 +2969,39 @@ elements of `a` are equal to 1, otherwise the behavior is undefined.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.tuple
+
+### Semantics
+Groups a variadic number of tokens, tuples, or tensors in `val` into a tuple,
+`result`.
+
+### Inputs
+
+| Name  | Type                                                                |
+|-------|---------------------------------------------------------------------|
+| `val` | variadic number of tokens, tuples, or tensors of any supported type |
+
+### Outputs
+
+| Name     | Type                                                                       |
+|----------|----------------------------------------------------------------------------|
+| `result` | a variadic sized tuple of tokens, tuples, or tensors of any supported type |
+
+### Constraints
+
+  * (C1) For all i, `type(val[i])` = `type(result[i])`.
```

Comment:
Feedback:
* Let's introduce another constraint to use N for (C2)
* It may be inappropriate to index `val` using `[i]` since it is variadic. How about:
```
* (C1) size(`val`) $=$ size(`result`) $=$ N.
* (C2) For all `val_i` in `val`, type(`val_i`) $=$ `result[i]` for all `i` $\in$ range [0, N).
```

You can keep this thread open for the next reviewer.